### DISCUSSION TITLE
Is plugin causing GetDownloadSizeAsync always to be 0?

### DISCUSSION THREAD

**Discussion by a-l-e-x-d-s-9** [jelte/be.khepri.play.assetdelivery.addressables#17] [General]
I'm not sure if it is related to the plugin, or maybe I configured something wrong, or misunderstood something.
I'm trying to check if Addressable bundle already downloaded, for on-demand bundle in internal test, by using Addressables.GetDownloadSizeAsync to check if size is 0,  and it always does. Even if I launch the game for the first time, and bundle is definitely wasn't downloaded. Also, if I clear the storage and cache for the game, when I launch the game, GetDownloadSizeAsync is 0. 
Am I wrong that GetDownloadSizeAsync should not be 0 for bundles that weren't downloaded yet? 

My ~code:
``` script
private IEnumerator asyncInit()
{
    yield return Addressables.InitializeAsync();
    AsyncOperationHandle< long > getDownloadSize = Addressables.GetDownloadSizeAsync( _adressable_reference.RuntimeKey );
    yield return getDownloadSize;
    if ( 0 == getDownloadSize.Result )
    {
        Debug.Log( "GetDownloadSizeAsync: 0." );
    }
    else
    {
       Debug.Log( $"GetDownloadSizeAsync: {getDownloadSize.Result}." );
    }
}
```

[I took the code from here.](https://docs.unity3d.com/Packages/com.unity.addressables@1.16/manual/DownloadDependenciesAsync.html)

**Comment by jelte** [comment]
ah, yes probably should document this. the `Addressables.GetDownloadSizeAsync` checks if the path is remote. Because the plugin doesn't change anything in Addressables itself, for Addressables the bundle is not remote. As such, it always returns 0 when it computes the size. Changing this would require changes in the build process of Addressables and my make the package easily incompatible with future updates of Addressables. It is definitely something I would like to take a look at in the future.

For now you should use `AddressablesAssetDelivery.GetDownloadSizeAsync` in addition to `Addressables.GetDownloadSizeAsync`, 

```
private IEnumerator asyncInit()
{
    yield return Addressables.InitializeAsync();
    AsyncOperationHandle< long > getAddresablesDownloadSize= Addressables.GetDownloadSizeAsync( _adressable_reference.RuntimeKey );
    AsyncOperationHandle< long > getAssetPackDownloadSize = AddressablesAssetDelivery.GetDownloadSizeAsync( _adressable_reference.RuntimeKey );
    yield return getAddresablesDownloadSize;
    yield return getAssetPackDownloadSize;

    var downloadSize = getAddresablesDownloadSize.Result + getAssetPackDownloadSize.Result;
    Debug.Log( $"GetDownloadSizeAsync: {downloadSize}." );
}
```

**Reply by a-l-e-x-d-s-9** [reply]
Thanks!

**Reply by jelte** [reply]
I assume you mean which namespace you need to include?

```
using Khepri.AssetDelivery;
```
